test/xpu/test_gemm.py

```diff
@@ -1189,18 +1190,64 @@ def weight_int4pack_mm(a, b_int4pack, qscale, qzeros):
                 a, b_int4pack, q_group, qscale, qzeros
             )
 
+        def unpack_weight(qweight, scales, qzeros, q_config):
+            group_size = q_config["group_size"]
+            bits = q_config["bits"]
+            s32_bits = 32
+
+            assert bits == 4
+            # Int32 can store 8 * 4bits data. This is the offset for each data.
+            wf = (
+                torch.tensor(list(range(0, s32_bits, bits)), dtype=torch.int32)
+                .unsqueeze(0)
+                .to("xpu")
+            )
+            zeros = qzeros
+            if qzeros is not None:
+                zeros = zeros.to(torch.int8)
+
+            weight = torch.bitwise_right_shift(
+                torch.unsqueeze(qweight, 1).expand(-1, 32 // bits, -1), wf.unsqueeze(-1)
+            ).to(torch.int16 if bits == 8 else torch.int8)
+            torch.bitwise_and(weight, (2**bits) - 1, out=weight)
+
+            return weight, scales, zeros
+
+        def dequantize(qweight, scales, qzeros, group_size, g_idx=None):
+            q_config = {"group_size": group_size, "bits": 4}
+            weight, gptq_scales, gptq_zeros = unpack_weight(
+                qweight, scales, qzeros, q_config
+            )
+            # gptq_zeros = (torch.ones_like(gptq_zeros) * 8).to("xpu")  # hard code zp
+            if len(weight.shape) > 2:
+                weight = weight.reshape(-1, weight.shape[-1])
+            infeatures = weight.shape[0]
+            if g_idx is None:
+                g_idx = torch.tensor(
+                    [i // q_config["group_size"] for i in range(infeatures)],
+                    dtype=torch.int32,
+                )
+            if gptq_zeros is None:
+                return (weight - 8) * gptq_scales[g_idx]
+            else:
+                return (weight - gptq_zeros[g_idx]) * gptq_scales[g_idx]
+
         b_int4pack, b_scales, zeros_int8 = convert_weight_to_int4pack(b_bf16)
 
-        for dtype in [torch.bfloat16, torch.float16]:
-            a = a_bf16.to(dtype=dtype)
-            b = b_bf16.to(dtype=dtype)
-            b_scales = b_scales.to(dtype=dtype)
-            ref = torch.mm(a, b)
```

Comment:
Please do not change the test logic of int4 gemm ut.
1. GPTQ XPU is not supported at AO now. The ut is only intended for RTM recipe.
2. Using `dq` weight to construct ref is not recommended. We formerly found that it would hid bugs in weight prepack. 
3. The UT follows the usage of pytorch designed cases, we want keep a same logic.